pandas/tests/frame/methods/test_astype.py

```diff
@@ -714,6 +714,15 @@ def test_astype_noncontiguous(self, index_slice):
         expected = df.iloc[index_slice]
         tm.assert_frame_equal(result, expected, check_dtype=False)
 
+    def test_astype_retain_attrs(self):
```

Comment:
Could you check if we have tests for the other cases. If not, please parametrize here

Reply:
Just to clarify, we would want a case for changing all columns' data type correct?

Reply:
No, there are different options of parametrizing astype, a data type, data type represented as string or a dict of one of both

Reply:
Oh ok, I see, I will double-check for this and add any tests if necessary.

Reply:
Added the parametrization